src/update-done/update-done.c

```diff
@@ -50,20 +51,20 @@ static int apply_timestamp(const char *path, struct timespec *ts) {
         if (r < 0)
                 return log_error_errno(r, "Failed to set SELinux context for %s: %m", path);
 
-        fd = open(path, O_CREAT|O_WRONLY|O_TRUNC|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, 0644);
+        fd = open_tmpfile_linkable(path, O_CREAT|O_WRONLY|O_TRUNC|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW, &tmp);
         mac_selinux_create_file_clear();
 
         if (fd < 0) {
                 if (errno == EROFS)
-                        return log_debug("Can't create timestamp file %s, file system is read-only.", path);
+                        return log_debug("Can't create timestamp tempfile %s, file system is read-only.", tmp);
```

Comment:
"tempfile" is not an english word. Please say "temporary timestamp file" here...